pkg/infra/iac2/templates/iam_role/factory.ts

```diff
@@ -12,7 +12,7 @@ interface Args {
 // noinspection JSUnusedLocalSymbols
 function create(args: Args): aws.iam.Role {
     return new aws.iam.Role(args.Name, {
-        assumeRolePolicy: args.AssumeRolePolicyDoc,
+        assumeRolePolicy: JSON.stringify(args.AssumeRolePolicyDoc),
```

Comment:
One thing to note here is that if you don't do the stringify, then this `factory.ts` will still compile, but the eventual `index.ts` will fail. We should consider whether this is something we can guard against (either by finding an alternative index.ts-rendering that allows both formats, or by finding Go+TS representations that let us catch this in unit tests).

I think the only actionable thing I'd kinda like to see is:
* a brief writeup of what we're doing, why we're doing it, its limitations, and alternatives we considered (along with why we didn't do them)
* a ticket to consider revisiting the issue in the fit-and-finish milestone, with a link to that writeup. We can always punt on that ticket once we get to that milestone (and the ticket should say that, so we don't forget).

Basically, this is subtle enough that I think there's a decent chance we'd want to revisit it, and when we do there's a really good chance we won't remember why we did it this way unless we write it down.

Reply:
yeah where do you want us to track all of this? do you want me to just write a task for it in that milestone with all of the notes? 

Also regarding the internal documentation, is that in regards to how to use pulumi.apply for us or why we need to do json.stringify?

Reply:
I think a task is  fine in the milestone with all the notes there. If that becomes annoying to discuss (when we get to it), we can always convert it to a gdoc or to a GH Discussion.

I was thinking basically both the things you mentioned. But more specifically, I think this is probably all the info we'd want to capture (though obviously feel free to add anything else you think is relevant):

* why we need the pulumi.apply (not so much "how to use it", but really "why did we need to develop all this in the first place")
* and then the fact that we wrap it in the `pulumi.apply` (so I guess, the "how" from bullet 1)
* and then why we need the stringify, what will go wrong if we don't include it, and specifically calling out that this means we lose the guarantee of "if factory.ts compiles and unit tests pass, then index.ts will also compile"
* any alternatives we considered (I think it was just inlining the pulumi.interpolate?), and why those didn't work

Reply:
created https://github.com/orgs/klothoplatform/projects/1/views/18?pane=issue&itemId=24930378 with details